### PR TITLE
Neo Address Resolution

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Comment by steven1227** [comment]
Why not put Neo addresses in NNS directly?

**Comment by erikzhang** [comment]
> Why not put Neo addresses in NNS directly?

As I've answered in https://github.com/neo-project/neo/issues/2284#issuecomment-778574349:

> My design philosophy is to design a lightweight core as much as possible. For problems that can be solved with standards, try not to add functions to the core. Unless this can bring very obvious advantages, such as performance improvements.

**Review by shargon — APPROVED** [review]

File: nep-18.mediawiki [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+We use <code>address</code> as the attribute name of the Neo address:
+
+<pre>
+"address=<Neo address>"
+</pre>
+
+The following TXT record map the domain name to the Neo address <code>NUVPACMnKFhpuHjsRjhUvXz1XhqfGZYVtY</code>:
+
+<pre>
+"address=NUVPACMnKFhpuHjsRjhUvXz1XhqfGZYVtY"
```
Comment (shargon) [inline nep-18.mediawiki:38]: Maybe we should specify the address type, like `address.neo=`

Comment (erikzhang) [reply nep-18.mediawiki:38]: Do we need to support multiply records for each type?

Comment (shargon) [reply nep-18.mediawiki:38]: Or separated by ';'

Comment (erikzhang) [reply nep-18.mediawiki:38]: But RFC 1464 doesn't allow `;`.

Comment (fyrchik) [reply nep-18.mediawiki:38]: I think multiple records is the way to go: if a single record contains a single address, calling by name from inside the contract is much easier (parsing base58 vs splitting a string + parsing). Also multiple records are helpful with other usages for the TXT record.

Comment (erikzhang) [reply nep-18.mediawiki:38]:
```
neo.xxx.xx TXT "address=NUVPACMnKFhpuHjsRjhUvXz1XhqfGZYVtY"
btc.xxx.xx TXT "address=14qViLJfdGaP4EeHnDyJbEGQysnCpwn1gd"
```

**Comment by roman-khimov** [comment]
NEP-1 as it is now:

> If the NEP collaborators approve, the NEP editor will assign the NEP a number, label it as Standards Track, Informational, or Meta, give it status "Draft", and add it to the git repository.

> Once the NEP is ready for the repository, the NEP editor will: 
> ...
> Merge the pull request...
> List the NEP in [README.mediawiki](https://github.com/neo-project/proposals/blob/master/README.mediawiki)

But we have this one, #20 and #21 with an "Accepted" status, listed in README and yet not merged into the repository. Is this correct from the process POV? I think we should merge them at least. If any changes needed they can be done in separate PRs until these NEPs reach "Final" status. And then "Final" NEPs MUST NOT be touched except for grammar, spelling, or markup mistakes. Maybe we want to update NEP-1 with this?

**Comment by Jim8y** [comment]
@shargon should we merge the accepted proposal?

**Comment by shargon** [comment]
> @shargon should we merge the accepted proposal?

Yes, we are all agree? please vote with 👍  or 👎

**Comment by roman-khimov** [comment]
We will provide some examples afterwards, we're storing contract hashes in NNS already.

**Review by shargon — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]

**Review by vncoelho — COMMENTED** [review]
And where is the domain name?

**Comment by vncoelho** [comment]
This is just for the address themself? no directly relation to the domain that will be linked, right?

**Review by shargon — COMMENTED** [review]
@roman-khimov maybe we should use `address.N3=` ?

**Comment by roman-khimov** [comment]
I don't think it's worth the trouble. Legacy is almost non-existent to me and it's easily separated from N3 because of `A` prefix.

**Comment by shargon** [comment]
> I don't think it's worth the trouble. Legacy is almost non-existent to me and it's easily separated from N3 because of `A` prefix.

But you can store ethereum address o bane address also...

**Comment by cschuchardt88** [comment]
I think this should function like DNS. 

> > I don't think it's worth the trouble. Legacy is almost non-existent to me and it's easily separated from N3 because of `A` prefix.
> 
> But you can store ethereum address o bane address also...

The `hostname` for the `txt` record would be whatever format the validator wants.

Example
---
| hostname | Record Type | Value |
| -------- | -------------- | --------- |
| ether-challenge | TXT | "address=0x26FfB57236FeA81746c9a1f2604fEB7ec1Bb4BDC" |

**Comment by roman-khimov** [comment]
> But you can store ethereum address o bane address also...

The way the NEP author wanted it is https://github.com/neo-project/proposals/pull/133#issuecomment-1999614522. To me it's acceptable, having this standard as it is now it better than not having it at all.

**Comment by superboyiii** [comment]
Better than nothing. Let's vote. @shargon @roman-khimov @AnnaShaleva @Jim8y @shargon @AnnaShaleva

**Comment by Jim8y** [comment]
Man, you have to tell us how to vote @superboyiii

**Review by vncoelho — COMMENTED** [review]
How to vote neutral as well? For me this is kind of neutral.

**Comment by Jim8y** [comment]
> How to vote neutral as well? For me this is kind of neutral.

 yes or no man please, decisions must be made.

**Comment by Jim8y** [comment]
But isn't this one already marked as accepetd?

**Comment by cschuchardt88** [comment]
After this can we make reverse lookup for new type `PTR`

**Review by Jim8y — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]
Definitely worth merging, it might be helpful for NeoFS.

**Review by cschuchardt88 — APPROVED** [review]

**Pull request merged**